src/main/kotlin/org/reactome/server/tools/WriteBioPAX3.kt

```diff
@@ -0,0 +1,250 @@
+@file:Suppress("ktlint:standard:no-wildcard-imports")
+
+package org.reactome.server.tools
+
+import org.biopax.paxtools.io.SimpleIOHandler
+import org.biopax.paxtools.model.*
+import org.biopax.validator.BiopaxIdentifier
+import org.biopax.validator.api.Validator
+import org.biopax.validator.api.ValidatorUtils
+import org.biopax.validator.api.beans.Validation
+import org.reactome.server.graph.domain.model.*
+import org.springframework.context.support.ClassPathXmlApplicationContext
+import java.io.*
+
+/**
+ * @author Sarah Keating <skeating@ebi.ac.uk>
+ */
+class WriteBioPAX3 {
+    companion object {
+        var validator: Validator? = null
+        var dbVersion: Int = 0
+        var xmlBase: String = ""
+
+        init {
+            try {
+                // Add JVM arguments to open required modules
+                System.setProperty("java.vm.vendor", "Oracle Corporation")
+                System.setProperty("java.vm.name", "Java HotSpot(TM) 64-Bit Server VM")
+                System.setProperty("java.vm.version", "17")
+                System.setProperty("java.vm.specification.version", "17")
+                System.setProperty("java.vm.specification.vendor", "Oracle Corporation")
+                System.setProperty("java.vm.specification.name", "Java Virtual Machine Specification")
+
+                val ctx =
+                    ClassPathXmlApplicationContext(
+                        "classpath:META-INF/spring/appContext-validator.xml",
+                        "classpath:META-INF/spring/appContext-loadTimeWeaving.xml",
+                    )
+                validator = ctx.getBean("validator") as Validator
+                println("BioPAX Validator initialized.")
+            } catch (e: Exception) {
+                System.err.println("Error initializing BioPAX Validator: ${e.message}")
+                e.printStackTrace()
+            }
+        }
+    }
+
+    private val thisPathway: Pathway?
+    private val thisListEvents: List<Event>?
+    private var parentPathway: Pathway? = null
+    private var stepNum: Int = 0
+    private var pathwayNum: Int = 0
+    private var biornNum: Int = 0
+    private var catNum: Int = 0
+    private var useEventOf: Boolean = true
+    private val bioPAXFactory = BioPAXLevel.L3.defaultFactory
+    private var thisModel: Model? = null
+
+    /**
+     * Construct an instance of the WriteBioPAX3
+     */
+    constructor() {
+        thisPathway = null
+        thisListEvents = null
+        parentPathway = null
+        xmlBase = "http://www.reactome.org/biopax/#"
```

Comment:
Maybe pull the base URL into a constant field "http://www.reactome.org/biopax/" so it's in one place.  Also can make it "https"